tests/test_rules.py

```
@@ -1317,3 +1317,81 @@ def test_command_sections_partial(temp_dir):
+
+    def test_unset_returns_schema_default(self):
+        rule = self._rule({})
+        assert rule.setting("max-tokens") == rule.config_schema["max-tokens"]["default"]
+
+    def test_explicit_null_returns_schema_default(self):
+        rule = self._rule({"max-tokens": None})
+        assert rule.setting("max-tokens") == rule.config_schema["max-tokens"]["default"]
+
+    def test_undeclared_name_raises_keyerror_naming_rule_and_option(self):
+        rule = self._rule({})
+        with pytest.raises(KeyError, match="content-section-length.*no-such-option"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)